python/pyarrow/tests/test_io.py

```diff
@@ -710,6 +710,15 @@ def test_non_cpu_buffer(pickle_module):
     # Sliced buffers with same address
     assert buf_on_gpu_sliced.equals(cuda_buf[2:4])
 
+    # Testing copying buffer
+    mm = ctx.memory_manager
+    for dest in [mm, mm.device]:
+        buf2 = cuda_buf.copy(dest)
```

Comment:
```suggestion
        buf2 = buf_on_gpu.copy(dest)
```

This test is generally testing with the `buf_on_gpu` object, because it is testing the base class `Buffer` class instances with a non-cpu device (in `test_cuda.py` we are testing `CudaBuffer`).